### ISSUE TITLE
Every download fails with HTTP 401 error

### ISSUE THREAD

**Issue by ddiggles** [TeamNewPipe/NewPipe#5773]
### Checklist

- [x] I am using the latest version - 0.20.11 
- [x] I checked, but didn't find any duplicates (open OR closed) of this issue in the repo. 
- [x] I have read the contribution guidelines given at https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md.
- [x] This issue contains only one bug. I will open one issue for every bug report I want to file.

### Steps to reproduce the bug

1) Open any video
2) Press "Download"
3) Select any size, press OK
4) Download starts then immediately fails with "Download failed" notification.

### Actual behaviour

Every video download fails.  In downloads list, error download reads "HTTP 401."

### Expected behavior

Videos were downloading just fine until a few days ago.

### Logs

No logs

### Device info

 - Android version/Custom ROM version: Android 8.1.0
 - Device model: Nexus 7 2013 wifi

**Comment by XiangRongLin** [comment]
works for me

**Comment by moriel5** [comment]
This is happening to me only on specific videos.
1. (Level1Techs) Icy Dock EZ-Adapter: U.2 to USB Made Easy!
2. (ShortCircuit) Perfectly balanced as all things should be - LG Gram 2021 Laptops

**Comment by XiangRongLin** [comment]
@moriel5 Then please adjust the initial post with that information. You wrote that it doesn't work on all videos, which is not the case

**Comment by ddiggles** [comment]
@XiangRongLin it is the case for me; I am the OP.  But looks like @moriel5 was able to reproduce only for specific videos.

**Comment by moriel5** [comment]
@XiangRongLin, as @ddiggles has already said, they are the OP, not me. I merely added my own findings.

Additionally for me, it does download, but stops and errors out every 0.5~1MB for me, so downloading a 300MB clip (I'm downloading 1440p) takes me around 350~550 clicks to resume (I had tested with both of the clips I had mentioned.

**Comment by ddiggles** [comment]
@moriel5 I can confirm this is the actual behavior as well.  I can resume but it errors again as soon as a tiny bit is downloaded.

**Comment by ChiefMikeK** [comment]
A couple of things to try.  
* backup your settings. 
* clear your phone's cache and restart your phone. 
* turn off your router/WiFi wait 30 seconds(takes time 2 drain to capacitor) then restart your router.
* scan for viruses. 
* ask for help @reddit
  :neckbeard:

**Comment by moriel5** [comment]
@ChiefMikeK As someone who manages the network at home, I can say that from my testing, the issues are not on the user's end (unless YouTube's website has changed yet again), and this won't fix the problem (it may solve other issues, however that is a different matter).

**Comment by moriel5** [comment]
Update: This is also happening with the Linus Tech Tips clip: The World's Most Expensive Xbox Controller is DONE.

**Comment by ChiefMikeK** [comment]
:-( Just saying that because 
- I've had malware that has caused google to think that I am a bot attacking them also had my IP is on some DNS blacklists.

**Comment by moriel5** [comment]
Fair point, @ChiefMikeK, though I can guarantee that everything on our network is clean (network admins, I feel you, and will soon be joining you, once I get through my CCNA and CCNP courses), and we are under no blacklist (the only possible reason why we might be blacklisted would be due to me employing DoH and DoT via CloudFlare, in conjunction to blocking ads and trackers(and even then, it would not be acceptible, since I do so moderately)).

**Comment by personaaleatoria** [comment]
I am seeing the same behavior although I couldn't reproduce with the videos @moriel5 commented about. I'm also not seeing a 401 error, the download just stops midway and it reads "(paused) Error" next to the download size number having downloaded just a few megabytes.

I've had some videos that fail at first and later on the download works for the same video. To my experience the error appears to occur randomly. When I use a VPN the download fails and recovers regardless of location.

I can only recommend that the Newpipe client disguises itself as much as possible as a common browser when making requests.

Newpipe 0.20.11

**Comment by makeyeu** [comment]
Can reproduce. Download fails on all videos. When clicked "start", download will resume. As mentioned above,  downloading bigger files takes dozens of clicks.

**Comment by moriel5** [comment]
I can no longer reproduce this, even with the videos I had this issue with, so I can only chalk this up to an issue, either on YouTube's end, our ISP's network, or the CDN in my country.

@personaaleatoria For me it showed the same, the 401 error was displayed when I checked the error details (overflow menu from the download).

**Comment by ddiggles** [comment]
@moriel5 Can confirm.  The problem seems to have spontaneously disappeared.
